File: app/src/main/java/com/example/expensetracker/ui/components/BudgetProgressBar.kt
package com.example.expensetracker.ui.components

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import com.example.expensetracker.ui.theme.EmeraldPrimary
import com.example.expensetracker.ui.theme.SlatePale

@Composable
fun BudgetProgressBar(
    spent: Double,
    budget: Double,
    modifier: Modifier = Modifier
) {
    val progress = if (budget > 0) (spent / budget).toFloat().coerceIn(0f, 1f) else 0f
    val isOverBudget = spent > budget

    Column(modifier = modifier) {
        LinearProgressIndicator(
            progress = { progress },
            modifier = Modifier
                .fillMaxWidth()
                .height(12.dp)
                .clip(RoundedCornerShape(6.dp)),
            color = if (isOverBudget) MaterialTheme.colorScheme.error else EmeraldPrimary,
            trackColor = SlatePale,
        )
        
        Spacer(modifier = Modifier.height(8.dp))
        
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Text(
                text = "Spent: $${String.format("%.2f", spent)}",
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurface
            )
            Text(
                text = "Budget: $${String.format("%.2f", budget)}",
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurface
            )
        }
    }
}

File: lib/widgets/budget_progress_bar.dart
import 'package:flutter/material.dart';
import 'package:expense_tracker_flutter/theme/app_colors.dart';
import 'package:provider/provider.dart';
import 'package:expense_tracker_flutter/providers/settings_provider.dart';

class BudgetProgressBar extends StatelessWidget {
  final double spent;
  final double budget;

  const BudgetProgressBar({
    Key? key,
    required this.spent,
    required this.budget,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final progress = budget > 0 ? (spent / budget).clamp(0.0, 1.0) : 0.0;
    final isOverBudget = spent > budget;

    final settings = context.watch<SettingsProvider>();

    return Column(
      children: [
        ClipRRect(
          borderRadius: BorderRadius.circular(6),
          child: LinearProgressIndicator(
            value: progress,
            minHeight: 12,
            backgroundColor: AppColors.slatePale,
            valueColor: AlwaysStoppedAnimation<Color>(
              isOverBudget
                  ? Theme.of(context).colorScheme.error
                  : AppColors.emeraldPrimary,
            ),
          ),
        ),
        const SizedBox(height: 8),
        Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          children: [
            Text(
              settings.hideBalances
                  ? 'Spent: ****'
                  : 'Spent: ${settings.currency}${spent.toStringAsFixed(2)}',
              style: Theme.of(context).textTheme.bodySmall,
            ),
            Text(
              settings.hideBalances
                  ? 'Budget: ****'
                  : 'Budget: ${settings.currency}${budget.toStringAsFixed(2)}',
              style: Theme.of(context).textTheme.bodySmall,
            ),
          ],
        ),
      ],
    );
  }
}
